definitions/types/gcp-grn.json

```diff
@@ -4,12 +4,14 @@
   "title": "GCP Resource Name (GRN)",
   "description": "GCP Resource Name (GRN)",
   "examples": [
-    "projects/my-project/global/networks/my-network",
-    "projects/my-project/locations/us-west2/instances/my-instance",
-    "projects/my-project/locations/us-west2/clusters/my-cluster",
-    "projects/my-project/zones/us-west2-a/instances/node-pool-id"
+    "projects/my-project/global/networks/my-global-network",
+    "projects/my-project/regions/us-west2/subnetworks/my-subnetwork",
+    "projects/my-project/topics/my-pubsub-topic",
+    "projects/my-project/subscriptions/my-pubsub-subscription",
+    "projects/my-project/locations/us-west2/instances/my-redis-instance",
+    "projects/my-project/locations/us-west2/clusters/my-gke-cluster"
   ],
-  "pattern": "^projects\/[a-z0-9-]+\/[a-z]+\/[a-z0-9-]+\/[a-z0-9-]+.*$",
+  "pattern": "^projects\/[a-z0-9-]+\/[a-z]+\/[a-z0-9-]+(?(?=\/)[a-z0-9-]+|.*)$",
```

Comment:
this fixes pubsub topics and subscriptions, they are 4 long not 5 or 6 like the others.

Reply:
I went through every bundle, and added examples + tested the regex. For the ones without grns, I added types and updated those art-defs.

Reply:
I think that last `|.*` kinda nullifies the type before it since it basically says "or any number of any characters"

Reply:
Okay I spent 15 more minutes trying to get better regex but I wasn't able to figure it out. Here I was trying to 

- if there's a forward slash `\/`, it has to be followed by our pattern `[a-z0-0-]` pattern.
- I added the `.*` in the else because that works for the grns that are 4, 5, 6 groups long.

I tried to nest the if / else to repeat that logic but didn't have much luck with that.

Reply:
`"^projects\/[a-z0-9-]+\/[a-z]+\/[a-z0-9-]+(?:\/[a-zA-Z0-9-_\.]+){0,6}$"`

Maybe just something like that ^
It covers the `projects/<project-name>/<service>/<service_name>` and then 0-6 blocks of `/` followed by any set of letters, numbers, `-`, `_` or `.`

That may be a bit loose very **every** service (perhaps some don't allow underscores or dots), but I don't think we need to get that specific.  I tested it on the examples you have listed, and all of them pass:

```
projects/my-project/global/networks/my-global-network
projects/my-project/regions/us-west2/subnetworks/my-subnetwork
projects/my-project/topics/my-pubsub-topic
projects/my-project/subscriptions/my-pubsub-subscription
projects/my-project/locations/us-west2/instances/my-redis-instance
projects/my-project/locations/us-west2/clusters/my-gke-cluster
```